### ISSUE TITLE
conda sends bad HTTP headers when querying noarch repos

### ISSUE THREAD

**Issue by ikalev** [conda/conda#4322]
``conda 4.3`` requires package repositories to provide the ``noarch`` platform directory. After adding ``noarch`` to my custom repository served by Microsoft ``IIS 7.5``, conda started failing after a few runs of ``conda install``:

```
$ conda install conda
Fetching package metadata ....
CondaHTTPError: HTTP 400 BAD REQUEST for url <http://myrepo/packages/noarch/repodata.json>
An HTTP error occurred when trying to retrieve this URL.
HTTPError(u'400 Client Error: Bad Request for url: http://myrepo/packages/noarch/repodata.json',)
```

Inspecting the HTTP headers sent by conda, it seems that the ETag ``If-None-Match`` contains bad characters:

```
$ conda install conda --debug
DEBUG conda.core.index:fetch_repodata(368): Locally invalidating cached repodata for http://myrepo/packages/noarch at C:\Anaconda2\pkgs\cache\8beb19b1.json
DEBUG requests.packages.urllib3.connectionpool:_make_request(400): http://dproxy:8080 "GET http://myrepo/packages/noarch/repodata.json HTTP/1.1" 400 311
DEBUG conda.core.index:fetch_repodata_remote_request(193):
>>GET /packages/noarch/repodata.json HTTP
> User-Agent: conda/4.3.4 requests/2.12.4 CPython/2.7.13 Windows/7 Windows/6.1.7601
> Accept: */*
> Accept-Encoding: gzip, deflate, compress, identity
> Connection: keep-alive
> Content-Type: application/json
> If-Modified-Since: Mon, 16 Jan 2017 15:27:55 GMT
> If-None-Match: \"b7eece1bd70d21:0\"

<<HTTP 400 Bad Request
< Content-Type: text/html; charset=us-ascii
< Date: Mon, 16 Jan 2017 15:30:47 GMT
< Proxy-Connection: Keep-Alive
< Server: Microsoft-HTTPAPI/2.0
< Content-Length: 311
< Elapsed: 00:00.006000

```
As you can see, the value of ``If-None-Match`` contains extra ``\"``, which IIS interprets as invalid. Stripping out these characters makes the request go through the server without any problems. 

Note that this happens only when conda queries the ``noarch`` subdirectory. All other platform subdirs work fine. Also note that this does not happen immediately; it takes a few retries until conda starts sending the malformed header.

**Comment by kalefranz** [comment]
Thanks for the report.  I've tracked down the problem.  Working on the solution now.

**Comment by kalefranz** [comment]
Actually, this was a big bug.  Thanks for the help in pointing it out.  Conda wasn't getting any correct 304s if the ETag header was set.  #4325 is the fix.

**Comment by kalefranz** [comment]
Should be fixed with #4325 in conda `4.3.5`.  If you'd like to test,

    conda config --add channels conda-canary
    conda update conda

**Comment by ikalev** [comment]
Thanks!
